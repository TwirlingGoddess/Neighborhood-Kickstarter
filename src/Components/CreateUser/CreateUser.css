.create-user-section {
  align-items: center;
  display: flex;
  height: 100vh;
  width: 100%;
  justify-content: center;
}

h1 {
  color: white;
}

.create-user-form {
  display: flex;
  align-items: center;
  border-radius: 0.7em;
  background-image: linear-gradient(to top, #09203f 0%, #537895 100%);
  justify-content: flex-start;
  width: 30rem;
  height: 28rem;
  flex-direction: column;
}

.input-fields {
  margin: .3rem;
  color: black;
  padding-left: .4rem;
  width: 20rem;
  height: 2.5rem;
  font-size: 1.5rem;
  background-image: linear-gradient(-225deg, #FFFEFF 0%, #D7FFFE 100%);
  border-radius: 0.6em;
  border: none;
}

.select-input {
  width: 20.2rem;
  margin-top: .5rem;
  font-size: .7rem;
  border-radius: 0;
}

.sign-up-button {
  box-sizing: border-box;
  appearance: none;
  background-color: transparent;
  border-radius: 0.6em;
  cursor: pointer;
  align-self: center;
  font-size: 1rem;
  font-weight: 400;
  line-height: 1;
  margin-top: 25px;
  padding: 1em 1em;
  text-decoration: none;
  text-align: center;
  text-transform: uppercase;
  font-family: 'Montserrat', sans-serif;
  font-weight: 700;
  border-color: #3498db;
  color: #fff;
  box-shadow: 0 0 40px 40px #3498db inset, 0 0 0 0 #3498db;
  transition: all 150ms ease-in-out;
  width: 19.8rem;
  max-height: 4rem;
}

.sign-up-button:hover {
  color: #fff;
  outline: 0;
  box-shadow: 0 0 10px 0 #3498db inset, 0 0 10px 4px #3498db;
}