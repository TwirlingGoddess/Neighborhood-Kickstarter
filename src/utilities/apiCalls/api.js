var fs = require('fs');
var S3FS = require('s3fs');
var s3fsImpl = new S3FS('leeleestestbucket1', {
  accessKeyId: 'AKIAIXU4Q4Y4XZWUK4SQ', 
  secretAccessKey: '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'
});

s3fsImpl.create();

var multiparty = require('connect-multiparty'),
  multipartyMiddleware = multiparty();

module.exports = function(router, passport){
  router.use(multipartyMiddleware);

  router.use(passport.authenticate('bearer', { session: false}));
  router.use(function(req, res, next){
    fs.appendFile('log.txt' req.path + " token: " + req.query.access_token + " ")
      function(err){
        next();
      });
  });

  router.post('/testupload', function(req, res){
    var file = req.files.file;
    var stream = fs.createReadStream(file.path);
    return s3fsImpl.writeFile(file.originalFilename, stream).then(function(){
      fs.unlink(file.path, function(err){
        if(err){
          console.error(err)
        }
      })
      res.redirect('/profile');
    })
  })

  router.get('/testAPI', function(req, res){
    res.json({ SecretData: 'abc123' });
  });


}