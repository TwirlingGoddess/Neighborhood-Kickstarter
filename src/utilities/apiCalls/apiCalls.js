export const getNeighborhoods = async () => {
  try {
    const response = await fetch('https://guarded-garden-68388.herokuapp.com/api/v1/neighborhoods');
    const neighborhoods = await response.json();
    return neighborhoods;
  } catch (error) {
    throw new Error(error.message)
  }
};

export const getProjects = async () => {
  try {
    const response = await fetch('https://guarded-garden-68388.herokuapp.com/api/v1/projects');
    const projects = await response.json();
    return projects;
  } catch (error) {
    throw new Error(error.message)
  }
};

export const getNeighborhoodProjectsById = async (id) => {
  try {
    const response = await fetch(`https://guarded-garden-68388.herokuapp.com/api/v1/neighborhoods/${id}`);
    const neighborhoodProjects = await response.json();
    return neighborhoodProjects;
  } catch (error) {
    throw new Error(error.message)
  }
};

export const getUserProjects = async (userId) => {
  try {
    const response = await fetch(`https://guarded-garden-68388.herokuapp.com/api/v1/users/${userId}/projects`);
    const userProjects = await response.json();
    return userProjects;
  } catch (error) {
    throw new Error(error.message)
  }
};

export const getAllUsers = async () => {
  try {
    const response = await fetch(`https://guarded-garden-68388.herokuapp.com/api/v1/users`);
    const users = await response.json();
    return users;
  } catch (error) {
    throw new Error(error.message)
  }
};

export const getProjectsComments = async (id) => {
  try {
    const response = await fetch(`https://guarded-garden-68388.herokuapp.com/api/v1/projects/${id}/comments`);
    const comments = response.json();
    return comments;
  } catch (error) {
    throw new Error(error.message)
  }
};

export const addNewUser = async (user) => {
  try {
    const response = await fetch(`https://guarded-garden-68388.herokuapp.com/api/v1/google_users`, {
      method: 'POST',
      headers: {
        'Content-type': 'application/json'
      },
      body: JSON.stringify(user)
    });
    const currentUser = await response.json();
    return currentUser;
  } catch (error) {
    throw new Error(error.message);
  }
};

export const addNewUserLocal = async (user) => {
  try {
    const response = await fetch('https://guarded-garden-68388.herokuapp.com/api/v1/users', {
      method: 'POST',
      headers: {
        'Content-type': 'application/json',
        'Accept': 'application/json'
      },
      body: JSON.stringify(user)
    });
    const currentUser = await response.json();
    return currentUser;
  } catch (error) {
    throw new Error(error.message);
  }
};

export const postNewProject = async (newProject, id) => {
  try {
    await fetch(`https://guarded-garden-68388.herokuapp.com/api/v1/users/${id}/projects`, {
      method: 'POST',
      headers: {
        'Content-type': 'application/json'
      },
      body: JSON.stringify(newProject)
    });
  } catch (error) {
    throw new Error(error.message);
  }
};

export const postNewComment = async (content, projectId) => {
  try { 
    await fetch(`https://guarded-garden-68388.herokuapp.com/api/v1/projects/${projectId}/comments`, {
      method: 'POST',
      headers: {
        'Content-type': 'application/json'
      },
      body: JSON.stringify(content)
    });
  } catch (error) {
    throw new Error(error.message);
  }
};

export const editPostedProject = async (project, id) => {
  try {
    await fetch(`https://guarded-garden-68388.herokuapp.com/api/v1/projects/${id}`, {
      method: 'PATCH',
      headers: {
        'Content-type': 'application/json'
      },
      body: JSON.stringify(project)
    });
  } catch (error) {
    throw new Error(error.message);
  }
};

export const sendEmailConfirmation = async (user) => {
  try {
    await fetch(`https://shrieking-chupacabra-95509.herokuapp.com/api/v1/notification`, {
      method: 'POST',
      headers: {
        'Content-type': 'application/json'
      },
      body: JSON.stringify(user)
    });
  } catch (error) {
    throw new Error(error.message);
  }
};