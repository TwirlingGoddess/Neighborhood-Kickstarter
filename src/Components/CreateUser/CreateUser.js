import React, { Component } from 'react';
import Select from 'react-select';
import { getNeighborhoods, addNewUserLocal, sendEmailConfirmation } from '../../utilities/apiCalls/apiCalls';
import './CreateUser.css';
import PropTypes from 'prop-types';

class CreateUser extends Component {
  constructor() {
    super();
    this.state = {
      allUsers: [],
      firstName: '',
      lastName: '',
      userName: '',
      email: '',
      password: '',
      neighborhood: '',
      neighborhoods: []
    };
  }

  componentDidMount = () => {
    this.setNeighborhoods();
  }

  setNeighborhoods = async() => {
    const response = await getNeighborhoods();

    const neighborhoods = response.map(neighborhood => {
      return { value: neighborhood.name, label: neighborhood.name, id: neighborhood.id };
    });

    this.setState({
      neighborhoods
    });
  }

  handleChange = (event) => {
    const { name, value } = event.target;

    this.setState({
      [name]: value
    });
  }

  handleSelectChange = (selectedOption) => {
    const neighborhood = selectedOption;

    this.setState({
      neighborhood
    });
  }

  handleSubmit = async (event) => {
    event.preventDefault();

    let {firstName, lastName, email, password, neighborhood, userName} = this.state;
    let { updateUser } = this.props;

    let localSignIn = {
      first_name: firstName,
      last_name: lastName,
      email,
      username: userName,
      district_id: neighborhood.id,
      password
    };

    let user = await addNewUserLocal(localSignIn);

    if (user) {
      let emailUser = {user_name: user.first_name, user_email: user.email};

      await sendEmailConfirmation(emailUser);

      updateUser(user);
      
      this.props.history.push('/Landing');
    }
  }

  render() {
    return (
      <div className='create-user-section'>
        <form className='create-user-form' onSubmit={this.handleSubmit}>
          <h1>Create Account</h1>
          <input
            className='input-fields first-name-input'
            name='firstName'
            type='text'
            value={this.state.firstName}
            placeholder='first name'
            onChange={this.handleChange}
          />
          <input 
            className='input-fields last-name-input'
            name='lastName'
            type='text'
            value={this.state.lastName}
            placeholder='last name'
            onChange={this.handleChange}
          />
          <input 
            className='input-fields'
            name='userName'
            type='text'
            value={this.state.userName}
            placeholder='user name'
            onChange={this.handleChange}
          />
          <input 
            className='input-fields email-input'
            name='email'
            type='email'
            value={this.state.email}
            placeholder='email'
            onChange={this.handleChange}
          />
          <input
            className='input-fields'
            name='password'
            type='text'
            value={this.state.password}
            placeholder='password'
            onChange={this.handleChange}
          />
          <Select
            className='select-input' 
            placeholder='Select your neighborhood'
            value={this.state.neighborhood}
            onChange={this.handleSelectChange}
            options={this.state.neighborhoods}
          />
          <button className='sign-up-button'>Sign Up</button>
        </form>
      </div>
    );
  }
}

CreateUser.propTypes = {
  history: PropTypes.array
};

export default CreateUser;