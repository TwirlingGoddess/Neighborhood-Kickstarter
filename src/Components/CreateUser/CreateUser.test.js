import React from 'react';
import CreateUser from './CreateUser';
import { getNeighborhoods, addNewUserLocal, sendEmailConfirmation } from '../../utilities/apiCalls/apiCalls';
import { shallow } from 'enzyme';

describe('CreateUser', () => {
  let wrapper;

  it('should match the snapshot', () => {
    wrapper = shallow(<CreateUser />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should have an initial state with the neighborhoods array', async () => {
    let selectedOptions = [];
    wrapper = shallow(<CreateUser />);
    wrapper.instance().setNeighborhoods(jest.fn());
    expect(wrapper.state('neighborhoods')).toEqual(selectedOptions);
  });

  it('should return data from getNeighborhoods fetch', async () => {
    const expected = [{"id": 24, "name": "Athmar Park"}, {"id": 1, "name": "Auraria"}, {"id": 70, "name": "Baker"}, {"id": 15, "name": "Barnum"}, {"id": 16, "name": "Barnum West"}, {"id": 13, "name": "Bear Valley"}, {"id": 3, "name": "Belcaro"}, {"id": 20, "name": "Berkeley"}, {"id": 50, "name": "Capitol Hill"}, {"id": 53, "name": "CBD"}, {"id": 22, "name": "Chaffee Park"}, {"id": 29, "name": "Cheesman Park"}, {"id": 7, "name": "Cherry Creek"}, {"id": 10, "name": "City Park"}, {"id": 61, "name": "City Park West"}, {"id": 52, "name": "Civic Center"}, {"id": 71, "name": "Clayton"}, {"id": 75, "name": "Cole"}, {"id": 60, "name": "College View - South Platte"}, {"id": 9, "name": "Congress Park"}, {"id": 2, "name": "Cory - Merrill"}, {"id": 8, "name": "Country Club"}, {"id": 78, "name": "DIA"}, {"id": 49, "name": "East Colfax"}, {"id": 69, "name": "Elyria Swansea"}, {"id": 65, "name": "Five Points"}, {"id": 12, "name": "Fort Logan"}, {"id": 58, "name": "Gateway - Green Valley Ranch"}, {"id": 66, "name": "Globeville"}, {"id": 44, "name": "Goldsmith"}, {"id": 32, "name": "Hale"}, {"id": 40, "name": "Hampden"}, {"id": 42, "name": "Hampden South"}, {"id": 47, "name": "Harvey Park"}, {"id": 59, "name": "Harvey Park South"}, {"id": 23, "name": "Highland"}, {"id": 30, "name": "Hilltop"}, {"id": 43, "name": "Indian Creek"}, {"id": 67, "name": "Jefferson Park"}, {"id": 39, "name": "Kennedy"}, {"id": 73, "name": "Lincoln Park"}, {"id": 57, "name": "Lowry Field"}, {"id": 48, "name": "Mar Lee"}, {"id": 11, "name": "Marston"}, {"id": 56, "name": "Montbello"}, {"id": 31, "name": "Montclair"}, {"id": 51, "name": "North Capitol Hill"}, {"id": 68, "name": "Northeast Park Hill"}, {"id": 33, "name": "North Park Hill"}, {"id": 37, "name": "Overland"}, {"id": 36, "name": "Platt Park"}, {"id": 21, "name": "Regis"}, {"id": 28, "name": "Rosedale"}, {"id": 38, "name": "Ruby Hill"}, {"id": 72, "name": "Skyland"}, {"id": 19, "name": "Sloan Lake"}, {"id": 41, "name": "Southmoor Park"}, {"id": 34, "name": "South Park Hill"}, {"id": 6, "name": "Speer"}, {"id": 55, "name": "Stapleton"}, {"id": 77, "name": "Sunnyside"}, {"id": 62, "name": "Sun Valley"}, {"id": 54, "name": "Union Station"}, {"id": 27, "name": "University"}, {"id": 46, "name": "University Hills"}, {"id": 35, "name": "University Park"}, {"id": 63, "name": "Valverde"}, {"id": 64, "name": "Villa Park"}, {"id": 45, "name": "Virginia Village"}, {"id": 4, "name": "Washington Park"}, {"id": 5, "name": "Washington Park West"}, {"id": 14, "name": "Washington Virginia Vale"}, {"id": 26, "name": "Wellshire"}, {"id": 17, "name": "West Colfax"}, {"id": 18, "name": "West Highland"}, {"id": 76, "name": "Westwood"}, {"id": 74, "name": "Whittier"}, {"id": 25, "name": "Windsor"}]
    let data = await getNeighborhoods();
    expect(data).toEqual(expected);
  });

  it('should update state with the input information', () => {
    let mockEvent = {target: {name: 'firstName', value: 'Bobby'}};
    let userInput = 'Bobby';
    wrapper = shallow(<CreateUser />);
    wrapper.instance().handleChange(mockEvent);
    expect(wrapper.state('firstName')).toEqual(userInput);
  });

  it('should update state with the neighborhood selected', () => {
    let selectedOption = {value: 'Auraria', label: 'Aurarua', id: 1};
    wrapper = shallow(<CreateUser />);
    wrapper.instance().handleSelectChange(selectedOption);
    expect(wrapper.state('neighborhood')).toEqual(selectedOption);
  });

  it('should update the input thats being typed in, in state', () => {
    let eventObject = { target: { name: 'firstName', value: 'a' } };
    wrapper = shallow(<CreateUser />);
    wrapper.find('.first-name-input').simulate('change', eventObject);
    expect(wrapper.state('firstName')).toEqual('a');
  });

  it('should return data from addNewUserLocal fetch', async () => {
    const userParam = {name: 'Lee', email: 'graham.la3@gmail.com'};
    const expected = {"message": "Incorrect parameters given!"};
    let data = await addNewUserLocal(userParam);
    expect(data).toEqual(expected);
  });

  it('should return data from sendEmailConfirmation fetch', async () => {
    const userParam = {name: 'Lee', email: 'graham.la3@gmail.com'};
    const expected = undefined;
    let data = await sendEmailConfirmation(userParam);
    expect(data).toEqual(expected);
  });

}); 