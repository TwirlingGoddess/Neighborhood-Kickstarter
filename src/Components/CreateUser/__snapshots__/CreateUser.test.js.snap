// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateUser should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CreateUser />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <form
        className="create-user-form"
        onSubmit={[Function]}
      >
        <h1>
          Create Account
        </h1>
        <input
          className="input-fields first-name-input"
          name="firstName"
          onChange={[Function]}
          placeholder="first name"
          type="text"
          value=""
        />
        <input
          className="input-fields last-name-input"
          name="lastName"
          onChange={[Function]}
          placeholder="last name"
          type="text"
          value=""
        />
        <input
          className="input-fields"
          name="userName"
          onChange={[Function]}
          placeholder="user name"
          type="text"
          value=""
        />
        <input
          className="input-fields email-input"
          name="email"
          onChange={[Function]}
          placeholder="email"
          type="email"
          value=""
        />
        <input
          className="input-fields"
          name="password"
          onChange={[Function]}
          placeholder="password"
          type="text"
          value=""
        />
        <StateManager
          className="select-input"
          defaultInputValue=""
          defaultMenuIsOpen={false}
          defaultValue={null}
          onChange={[Function]}
          options={Array []}
          placeholder="Select your neighborhood"
          value=""
        />
        <button
          className="sign-up-button"
        >
          Sign Up
        </button>
      </form>,
      "className": "create-user-section",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            Create Account
          </h1>,
          <input
            className="input-fields first-name-input"
            name="firstName"
            onChange={[Function]}
            placeholder="first name"
            type="text"
            value=""
          />,
          <input
            className="input-fields last-name-input"
            name="lastName"
            onChange={[Function]}
            placeholder="last name"
            type="text"
            value=""
          />,
          <input
            className="input-fields"
            name="userName"
            onChange={[Function]}
            placeholder="user name"
            type="text"
            value=""
          />,
          <input
            className="input-fields email-input"
            name="email"
            onChange={[Function]}
            placeholder="email"
            type="email"
            value=""
          />,
          <input
            className="input-fields"
            name="password"
            onChange={[Function]}
            placeholder="password"
            type="text"
            value=""
          />,
          <StateManager
            className="select-input"
            defaultInputValue=""
            defaultMenuIsOpen={false}
            defaultValue={null}
            onChange={[Function]}
            options={Array []}
            placeholder="Select your neighborhood"
            value=""
          />,
          <button
            className="sign-up-button"
          >
            Sign Up
          </button>,
        ],
        "className": "create-user-form",
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Create Account",
          },
          "ref": null,
          "rendered": "Create Account",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "input-fields first-name-input",
            "name": "firstName",
            "onChange": [Function],
            "placeholder": "first name",
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "input-fields last-name-input",
            "name": "lastName",
            "onChange": [Function],
            "placeholder": "last name",
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "input-fields",
            "name": "userName",
            "onChange": [Function],
            "placeholder": "user name",
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "input-fields email-input",
            "name": "email",
            "onChange": [Function],
            "placeholder": "email",
            "type": "email",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "input-fields",
            "name": "password",
            "onChange": [Function],
            "placeholder": "password",
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "className": "select-input",
            "defaultInputValue": "",
            "defaultMenuIsOpen": false,
            "defaultValue": null,
            "onChange": [Function],
            "options": Array [],
            "placeholder": "Select your neighborhood",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Sign Up",
            "className": "sign-up-button",
          },
          "ref": null,
          "rendered": "Sign Up",
          "type": "button",
        },
      ],
      "type": "form",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <form
          className="create-user-form"
          onSubmit={[Function]}
        >
          <h1>
            Create Account
          </h1>
          <input
            className="input-fields first-name-input"
            name="firstName"
            onChange={[Function]}
            placeholder="first name"
            type="text"
            value=""
          />
          <input
            className="input-fields last-name-input"
            name="lastName"
            onChange={[Function]}
            placeholder="last name"
            type="text"
            value=""
          />
          <input
            className="input-fields"
            name="userName"
            onChange={[Function]}
            placeholder="user name"
            type="text"
            value=""
          />
          <input
            className="input-fields email-input"
            name="email"
            onChange={[Function]}
            placeholder="email"
            type="email"
            value=""
          />
          <input
            className="input-fields"
            name="password"
            onChange={[Function]}
            placeholder="password"
            type="text"
            value=""
          />
          <StateManager
            className="select-input"
            defaultInputValue=""
            defaultMenuIsOpen={false}
            defaultValue={null}
            onChange={[Function]}
            options={Array []}
            placeholder="Select your neighborhood"
            value=""
          />
          <button
            className="sign-up-button"
          >
            Sign Up
          </button>
        </form>,
        "className": "create-user-section",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1>
              Create Account
            </h1>,
            <input
              className="input-fields first-name-input"
              name="firstName"
              onChange={[Function]}
              placeholder="first name"
              type="text"
              value=""
            />,
            <input
              className="input-fields last-name-input"
              name="lastName"
              onChange={[Function]}
              placeholder="last name"
              type="text"
              value=""
            />,
            <input
              className="input-fields"
              name="userName"
              onChange={[Function]}
              placeholder="user name"
              type="text"
              value=""
            />,
            <input
              className="input-fields email-input"
              name="email"
              onChange={[Function]}
              placeholder="email"
              type="email"
              value=""
            />,
            <input
              className="input-fields"
              name="password"
              onChange={[Function]}
              placeholder="password"
              type="text"
              value=""
            />,
            <StateManager
              className="select-input"
              defaultInputValue=""
              defaultMenuIsOpen={false}
              defaultValue={null}
              onChange={[Function]}
              options={Array []}
              placeholder="Select your neighborhood"
              value=""
            />,
            <button
              className="sign-up-button"
            >
              Sign Up
            </button>,
          ],
          "className": "create-user-form",
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Create Account",
            },
            "ref": null,
            "rendered": "Create Account",
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "input-fields first-name-input",
              "name": "firstName",
              "onChange": [Function],
              "placeholder": "first name",
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "input-fields last-name-input",
              "name": "lastName",
              "onChange": [Function],
              "placeholder": "last name",
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "input-fields",
              "name": "userName",
              "onChange": [Function],
              "placeholder": "user name",
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "input-fields email-input",
              "name": "email",
              "onChange": [Function],
              "placeholder": "email",
              "type": "email",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "input-fields",
              "name": "password",
              "onChange": [Function],
              "placeholder": "password",
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "className": "select-input",
              "defaultInputValue": "",
              "defaultMenuIsOpen": false,
              "defaultValue": null,
              "onChange": [Function],
              "options": Array [],
              "placeholder": "Select your neighborhood",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Sign Up",
              "className": "sign-up-button",
            },
            "ref": null,
            "rendered": "Sign Up",
            "type": "button",
          },
        ],
        "type": "form",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
